src/inmanta/ast/type.py

```diff
@@ -490,7 +499,11 @@ def function(*args, **kwargs):
         if len(args) != 1:
             raise NotImplementedError()
 
-        return expression.execute_direct({"self": args[0]})
+        try:
+            return expression.execute_direct({"self": args[0]})
+        except NotFoundException as e:
+            e.msg = "Unable to resolve `%s`: a type constraint has no access to the surrounding namespace." % e.stmt.name
```

Comment:
Is namespace the correct term?

Surrounding scope?

Can not reference variables?

Reply:
I think scope is indeed better than namespace. "Can not reference" variables is a result of this limitation, but might be more relevant to the user. I think that last one might be best then. Which has your preference?

Reply:
I follow your reasoning.